client/src/main/java/com/hiddenlayer/sdk/ModelScanService.java

```diff
@@ -116,7 +120,33 @@ public ScanReportV3 scanFolder(String folderPath, String modelName, OptionalInt
         out.flush();
         out.close();
         
-        return this.scanFile(tempFile.getAbsolutePath(), modelName, modelVersion);
+        return this.scanFile(tempFile.getAbsolutePath(), modelName, modelVersion, waitForDone);
+    }
+
+    public ScanReportV3 getDoneScanResults(UUID modelVersion) 
+        throws ApiException {
+          Double baseDelay = 100.0; // milliseconds
+          Integer retries = 0;
+
+          ScanReportV3 scanReport = this.getScanResults(modelVersion);
+  
+          while (scanReport == null 
+                  || (scanReport.getStatus() != ScanReportV3.StatusEnum.DONE 
+                        && scanReport.getStatus() != ScanReportV3.StatusEnum.FAILED
+                        && scanReport.getStatus() != ScanReportV3.StatusEnum.CANCELED)) {
+              retries += 1;
```

Comment:
Suggest capping the # of retries, rather than looping forever. Timeout should be really long since scans can wait for hours if the system is busy. Maybe 4-8 hours? Maybe later we could let the user configure that, but surely that's not in scope right now.